structure_tests/structure_test_v1.go

```diff
@@ -107,3 +120,45 @@ func (st StructureTestv1) RunFileContentTests(t *testing.T) {
 		}
 	}
 }
+
+// Given a list of environment variables and a list of lists of strings,
+// retrieve each environment variable's value and replace all occurrences
+// of it in each list of strings provided.
+func SubstituteEnvVars(t *testing.T, vars []string, lists ...*[]string) {
+	for _, env_var := range vars {
+		value := os.Getenv(env_var)
+		if value == "" {
+			t.Fatalf("Variable %s not found in environment!", env_var)
+		}
+		for _, list := range lists {
+			for i := range *list {
+				str := (*list)[i]
+				(*list)[i] = strings.Replace(str, "$"+env_var, value, -1)
+			}
+		}
+	}
+}
+
+func ProcessCommands(t *testing.T, commands []string) {
```

Comment:
this is how i originally wanted to do it, but the way i tried to implement it made some commands very difficult/maybe impossible to do. do you know how Kubernetes actually handles those commands?

golang's `exec.Cmd` takes two arguments: the top level command, and a list of flags to pass to it. so for a command like `install_node v5.9.0 -f`, it's as easy as splitting that string on spaces, passing the first string as the command and the rest as the flags.

but what if i want to do something like `sh -c 'apt-get update -qq && apt-get upgrade'`? i need that entire apt-get string to be one flag passed to `sh`...so that strategy totally falls through here. (this is one of the debian security tests btw)